### ISSUE TITLE
CodeMining that works on Photon does not work on 2018-09.

### ISSUE THREAD

**Issue by miklossy** [eclipse/xtext-eclipse#836]
CodeMining on Photon:
![codemining_photon_ok](https://user-images.githubusercontent.com/20393472/46279465-72057b80-c569-11e8-8288-8ac1588eb1b2.gif)

CodeMining on 2018-09:
![codemining_2018-09_nok](https://user-images.githubusercontent.com/20393472/46279473-7893f300-c569-11e8-9bd5-ac5ed3d47dc5.gif)

The reason is that the [AbstractXtextCodeMiningProvider](https://github.com/eclipse/xtext-eclipse/blob/master/org.eclipse.xtext.ui.codemining/src/org/eclipse/xtext/ui/codemining/AbstractXtextCodeMiningProvider.java#L144) creates a org.eclipse.jface.text.Position object setting only the offset but not the length.

With the [commit](https://github.com/eclipse/eclipse.platform.text/commit/d3f94a16e823f8bf57175e439539192c51c86267) on the eclipse.platform.text the behavior has been changed that all positions that's length is 0 considered as [deleted](https://github.com/eclipse/eclipse.platform.text/blob/master/org.eclipse.jface.text/src/org/eclipse/jface/text/source/inlined/InlinedAnnotationDrawingStrategy.java#L348).

The solution would be to modify the AbstractXtextCodeMiningProvider to set not only the offset, but also the length of the Position object.

**Comment by cdietrich** [comment]
We need tests

**Comment by miklossy** [comment]
It is quite difficult to cover this behaviour with junit tests, since the annotations are [draw](https://github.com/eclipse/eclipse.platform.text/blob/master/org.eclipse.jface.text/src/org/eclipse/jface/text/codemining/AbstractCodeMining.java#L146) on the StyledText canvas.

**Comment by kthoms** [comment]
Merged for 2.16

**Comment by cdietrich** [comment]
@miklossy 

with the greeting grammar

and

```
package org.xtext.example.mydsl1.ui.codemining

import org.eclipse.jface.text.BadLocationException
import org.eclipse.jface.text.IDocument
import org.eclipse.jface.text.codemining.ICodeMining
import org.eclipse.xtext.resource.XtextResource
import org.eclipse.xtext.util.CancelIndicator
import org.eclipse.xtext.util.IAcceptor
import org.xtext.example.mydsl1.myDsl.Model
import org.eclipse.xtext.nodemodel.util.NodeModelUtils
import org.xtext.example.mydsl1.myDsl.MyDslPackage

class MyDslCodeMiningProvider extends AbstractXtextCodeMiningProvider {
	override void createCodeMinings(IDocument document, XtextResource resource, CancelIndicator indicator,
		IAcceptor<? super ICodeMining> acceptor) throws BadLocationException {
		val Model m = resource.contents.head as Model
		var i = 0;
		for (g : m.greetings) {
			
			val o = NodeModelUtils.findNodesForFeature(g, MyDslPackage.Literals.GREETING__NAME).head.offset
			if (i%2==0)
			acceptor.accept(createNewLineContentCodeMining(o, "blablablablablablabla"))
			i++
		}
	}
}
``` 

i get 
![codeminings-bug2](https://user-images.githubusercontent.com/238736/46725776-2e072a80-cc7d-11e8-9616-114e9870c253.png)

could this be caused by your change?

**Comment by cdietrich** [comment]
=> either we should pass (1) or file a bug against codemining in platform for bad filtering

**Comment by cdietrich** [comment]
or should we extend api to pass a length too

**Comment by miklossy** [comment]
In order to be able to provide a proper fix for this problem, it would be useful to know how exactly the length parameter of a Position object is actually used by the code-mining framework. I think the corresponding code is located in the [org.eclipse.jface.text.source.AnnotationPainter](https://github.com/eclipse/eclipse.platform.text/blob/master/org.eclipse.jface.text/src/org/eclipse/jface/text/source/AnnotationPainter.java#L1404-L1427)

@angelozerr: Could you provide us some hints?
@cdietrich: Why do you think that setting the length parameter to a constant 1 value is sufficient?

**Comment by cdietrich** [comment]
maybe we should use 1 and or add overloads that allow to pass the length by the dsl developer

**Comment by angelozerr** [comment]
> Why do you think that setting the length parameter to a constant 1 value is sufficient?

Yes I do that too for JDT CodeMining and platform text Positions#of

**Comment by cdietrich** [comment]
Fixed in 2.16
